src/compiler/output-targets/copy/hashed-copy.ts

```diff
@@ -2,7 +2,38 @@ import { dirname, extname, join } from 'path';
 
 import type * as d from '../../../declarations';
 
-export const generateHashedCopy = async (config: d.ValidatedConfig, compilerCtx: d.CompilerCtx, path: string) => {
+/**
+ * Create a copy of a given file in its current directory but with a filename
+ * derived by hashing the file contents. This allows us to refer to a file by
+ * name and ensure that a given filename always refers to the same file
+ * contents.
+ *
+ * So for instance if you had a directory:
+ *
+ * build
+ * └── index.js
+ *
+ * and the contents of `build/index.js` hashed to `1234abcd` then write a
+ * file to `build/p-1234abcd.js` with the contents of `build/index.js`,
+ * giving:
+ *
+ * build
+ * ├── index.js
+ * └── p-1234abcd.js
+ *
```

Comment:
```suggestion
 * ```
 * build
 * ├── index.js
 * └── p-1234abcd.js
 * ```
```